File: code/go/day24/day24.go
package day24

import "../utils/linkedlist"

func Merge(l1, l2 *linkedlist.List) linkedlist.List {
	// 对输入的异常情况进行处理
	if l1 == nil && l2 == nil {
		return linkedlist.NewLinkedList()
	} else if l1 == nil {
		return *l2
	} else if l2 == nil {
		return *l1
	}

	p1 := l1.GetHead()
	p2 := l2.GetHead()
	newL := linkedlist.NewLinkedList()

	for {
		if p1 != nil &&  p2 != nil {
			val1 := p1.GetData().(int)
			val2 := p2.GetData().(int)

			if val1 > val2 {
				newL.Append(val2)
				p2 = p2.GetNext()
			} else {
				newL.Append(val1)
				p1 = p1.GetNext()
			}
		} else {
			break
		}
	}

	if p1 == nil {
		for p2 != nil {
			newL.Append(p2.GetData())
			p2 = p2.GetNext()
		}
	}

	if p2 == nil {
		for p1 != nil {
			newL.Append(p1.GetData())
			p1 = p1.GetNext()
		}
	}
	
	return newL
}

File: code/python/day24/day24.py
import unittest

from utils.linked_list import LinkedList, Node


def merge(ll1, ll2):
    ll = LinkedList()

    c1 = ll1.head
    c2 = ll2.head

    while c1 and c2:
        if c1.get_data() < c2.get_data():
            ll.append(c1.get_data())
            c1 = c1.get_next()
        else:
            ll.append(c2.get_data())
            c2 = c2.get_next()

    while c1:
        ll.append(c1.get_data())
        c1 = c1.get_next()

    while c2:
        ll.append(c2.get_data())
        c2 = c2.get_next()
    return ll


class TestMergeLinklist(unittest.TestCase):
    def test_1(self):
        ll1 = LinkedList()
        ll1.append(2)
        ll1.append(4)
        ll1.append(6)
        ll1.append(8)
        ll1.append(10)

        ll2 = LinkedList()
        ll2.append(1)
        ll2.append(3)
        ll2.append(5)
        ll2.append(7)
        ll2.append(9)
        ll = merge(ll1, ll2)
        ll.print()


if __name__ == '__main__':
    unittest.main()
